src/Core/Configurations/ConfigFileWatcher.cs

```diff
@@ -1,30 +1,18 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System.IO.Abstractions;
-using Azure.DataApiBuilder.Config;
-using Path = System.IO.Path;
-
 namespace Azure.DataApiBuilder.Core.Configurations;
 
 public class ConfigFileWatcher
 {
     private FileSystemWatcher? _fileWatcher;
     private readonly RuntimeConfigProvider? _configProvider;
 
-    public ConfigFileWatcher(RuntimeConfigProvider configProvider)
+    public ConfigFileWatcher(RuntimeConfigProvider configProvider, string directoryName, string configFileName)
```

Comment:
Is there any way to combine these into a single parameter? e.g. FullConfigFilePath? then the process of breaking that  down into a path + filename.extension can be handled as an implementation detail.

Reply:
unfortunately no, the directory name depends on the loader, so we'd have to pass the loader into the config file watcher which is less desirable than just combining these strings I reckon.